File: Tests/SkeletonPythonParserTests/Fixtures/sample.py
from typing import Optional, List


class Animal:
    name: str
    age: int

    def __init__(self, name: str, age: int) -> None:
        self.name = name
        self.age = age

    def greet(self, greeting: str) -> str:
        return f"{greeting}, {self.name}"

    def describe(self) -> str:
        return f"{self.name} ({self.age})"


class Dog(Animal):
    breed: str

    def __init__(self, name: str, age: int, breed: str) -> None:
        super().__init__(name, age)
        self.breed = breed

    def fetch(self, item: str) -> bool:
        return True


class Shape:
    def area(self) -> float:
        return 0.0

    def perimeter(self) -> float:
        return 0.0

File: Tests/SkeletonZigParserTests/Fixtures/sample.zig
const std = @import("std");

pub const Animal = struct {
    name: []const u8,
    age: u32,

    pub fn init(name: []const u8, age: u32) Animal {
        return Animal{ .name = name, .age = age };
    }

    pub fn greet(self: *const Animal, greeting: []const u8) []const u8 {
        _ = greeting;
        return self.name;
    }
};

pub const Shape = enum {
    circle,
    rectangle,
    triangle,

    pub fn describe(self: Shape) []const u8 {
        return switch (self) {
            .circle => "circle",
            .rectangle => "rectangle",
            .triangle => "triangle",
        };
    }
};

pub const Point = struct {
    x: f64,
    y: f64,

    pub fn distance(self: Point, other: Point) f64 {
        _ = other;
        return self.x;
    }
};

pub const Value = union {
    int_val: i64,
    float_val: f64,
    bool_val: bool,
};
